src/test/java/namor/template/stack/MapBasedStackTest.java

```diff
@@ -92,4 +92,35 @@ public void peakReturnsLastPushedElement() {
     public void whenPopFromEmptyStackThrowException() {
         mapBasedStack.pop();
     }
+
+    @Test
+    public void stackWorksWithFloat() {
+        ArrayStack<Float> stringArrayStack = new ArrayStack<>(3);
+        final float expected = 2.16f;
+        float actual = 4.4f;
+
+        stringArrayStack.push(1.1f);
+        stringArrayStack.push(2.16f);
+        stringArrayStack.push(3.3f);
+        stringArrayStack.pop();
+
+        if(!stringArrayStack.isEmpty()) {
+            actual = stringArrayStack.pop();
+        }
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    public void stackWorksWithChar() {
+        ArrayStack<Character> booleanArrayStack = new ArrayStack<>(3);
+        final char expected = 'a';
+        final char actual;
+
+        if(booleanArrayStack.isEmpty()) {
+            booleanArrayStack.push('A');
+            booleanArrayStack.push('a');
+        }
+        actual = booleanArrayStack.pop();
+        assertThat(actual).isEqualTo(expected);
+    }
 }
```

Comment:
get rid of useless tests in a sake of YAGNI (you're ain't gona need it - one of the principals of good desighn)